common/source/kernel/ProcessRegistry.cpp

```diff
@@ -65,9 +65,18 @@ void ProcessRegistry::Run()
   vfs.rootUmount();
 
   Scheduler::instance()->printStackTraces();
-
   Scheduler::instance()->printThreadList();
 
+  PageManager* pm = PageManager::instance();
+  if(!DYNAMIC_KMM && pm->getNumFreePages() != pm->getNumPagesForUser())
```

Comment:
This could cause problems for pages that are allocated later on by the OS for the kernel related things that do not need to be freed. For example for drivers, extra kernel mappings, a vmxon region,....

Reply:
Might need to e.g. add a tag parameter to the `allocPPN()`  function to track what a page is going to be used for.